hca/util/fs_helper.py

```diff
@@ -0,0 +1,80 @@
+import os, tempfile, errno
+from datetime import datetime
+
+
+class FSHelper(object):
+    """
+    Static helper class providing convenience methods for reading and writing files.
+    """
+
+    @staticmethod
+    def atomic_write(filename, shared_dir, content):
+        """
+        Atomically writes content a to file.
+
+        Writes to temp file and renames file (atomic operation) to desired filename.
+        To ensure atomicity, the src and dest dirs must be the same.
+
+        :param filename: File name only (e.g. example.txt)
+        :param shared_dir: Shared directory of temp and dest file
+        :param content: Content to write to file
+        """
+        dest_filename = os.path.join(shared_dir, filename)
+        try:
+            with tempfile.NamedTemporaryFile('wb', dir=shared_dir, delete=False) as temp:
+                temp_filename = temp.name
+                temp.write(content)
+            FSHelper._atomic_rename(temp_filename, dest_filename)
+        finally:
+            if temp_filename and os.path.isfile(temp_filename):
+                os.remove(temp_filename)
+
+    @staticmethod
+    def _atomic_rename(src, dest):
+        """
+        Atomically renames a file.
+        Support for Unix and Windows platforms.
+
+        :param src: Source filename
+        :param dest: Destination filename
+        """
+        max_retries = 5
+        for _ in range(max_retries):
+            try:
+                os.rename(src, dest)
+                return
+            except OSError as e:
+                FSHelper._handle_rename_error(e, src, dest)
+
+    @staticmethod
+    def _handle_rename_error(err, src, dest):
```

Comment:
I think you could just inline this method.  It's not particularly large and it's not called elsewhere.

Reply:
Yea I agree, however, it's pulled out now since when it was inline in `_atomic_rename`, it violated our linter's Code Complexity limit (7 > 6)